### PR TITLE
Replace sequelize with knex & objection

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: server/.eslintrc.yml [hunk 1]
```diff
@@ -18,3 +18,22 @@ rules:
       - info
   prefer-promise-reject-errors:
     - off
+  import/no-import-module-exports:
+    - off
+  import/no-relative-packages:
+    - off
```
Comment (what1s1ove) [inline server/.eslintrc.yml:24]: let's try avoid `off` and rewrite or turn off only in places where we really need it

File: server/src/data/db/connection.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import Sequelize from 'sequelize';
```
Comment (what1s1ove) [inline server/src/data/db/connection.js:1]: let's remove `connection.js` and do all this logic in the index file
ex: https://github.com/BinaryStudioAcademy/bsa-2021-jabber/blob/development/backend/src/server.ts#L19

File: server/src/data/models/base/base.model.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { Model, snakeCaseMappers } from 'objection';
+
+class BaseModel extends Model {
```
Comment (what1s1ove) [inline server/src/data/models/base/base.model.js:3]:
```suggestion
class Abstract extends Model {
```
don't forget about file name

File: README.md [hunk 4]
```diff
@@ -105,11 +106,11 @@
...
     Для того, чтобы указать ключи для Gyazo Storage, необходимо зарегистрироваться на сайте [Gyazo](https://gyazo.com/captures) и [зарегистрировать приложение](https://gyazo.com/oauth/applications). Затем в **.env**  использовать `access token` из только что созданного приложения в Gyazo.
 
-4. Выполните [миграции](https://sequelize.org/master/manual/migrations.html) и сиды для того, чтобы заполнить базу данных демо-данными. Для этого в командной строке (терминале) в папке server выполните:
+4. Выполните [миграции](https://knexjs.org/#Migrations) и сиды для того, чтобы заполнить базу данных демо-данными. Для этого в командной строке (терминале) в папке server выполните:
 
     ```
-    npx sequelize-cli db:migrate
-    npx sequelize-cli db:seed:all
+    npm run migrate:run
+    npm run seed:run
```
Comment (what1s1ove) [inline README.md:113]:
```suggestion
```

File: server/package.json [hunk 5]
```diff
@@ -6,6 +6,10 @@
     "npm": ">=8 <9"
   },
   "scripts": {
+    "migrate:make": "npx knex --esm migrate:make",
+    "migrate:rollback": "npx knex --esm migrate:rollback",
+    "migrate:run": "npx knex --esm migrate:latest",
+    "seed:run": "npx knex --esm seed:run",
```
Comment (what1s1ove) [inline server/package.json:12]:
```suggestion
```

File: server/src/data/seed-data/commentsSeed.js [hunk 6]
```diff
@@ -1,7 +1,4 @@
-/* eslint-disable max-len */
```
Comment (what1s1ove) [inline server/src/data/seed-data/commentsSeed.js:1]: don't forget remove seed folder

File: server/src/data/seeds/fill_db.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
+import { usersSeed, userImagesSeed } from '../seed-data/usersSeed';
```
Comment (what1s1ove) [inline server/src/data/seeds/fill_db.js:1]: delete seed

File: server/src/data/migrations/20211211160459_fill_db.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
+import { usersSeed, userImagesSeed } from '../seed-data/usersSeed';
```
Comment (what1s1ove) [inline server/src/data/migrations/20211211160459_fill_db.js:1]: Let's name them in kebab-case

File: server/src/data/migrations/20211211160459_fill_db.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+const ColumnName = {
+  IMAGE_ID: 'image_id',
+  IS_LIKE: 'is_like',
+  POST_ID: 'post_id',
+  USER_ID: 'user_id'
+};
+
+const randomIndex = length => Math.floor(Math.random() * length);
```
Comment (what1s1ove) [inline server/src/data/migrations/20211211160459_fill_db.js:20]: getRandomIndex

**Review by what1s1ove — APPROVED** [review]

File: server/src/data/migrations/20211208202419_create-tables.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+};
+
+export async function up(knex) {
+  await knex.schema.createTable(TableName.USERS, table => {
+    table.increments(ColumnName.ID).primary();
+    table.string(ColumnName.EMAIL).notNullable().unique();
+    table.string(ColumnName.USERNAME).notNullable().unique();
+    table.string(ColumnName.PASSWORD).notNullable();
+    table.dateTime(ColumnName.CREATED_AT).notNullable().defaultTo(new Date().toISOString());
+    table.dateTime(ColumnName.UPDATED_AT).notNullable().defaultTo(new Date().toISOString());
```
Comment (PavelSukhinin) [inline server/src/data/migrations/20211208202419_create-tables.js:28]: let's try `.defaultTo(knex.fn.now())` ?

File: server/src/data/models/abstract/abstract.model.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import { Model, snakeCaseMappers } from 'objection';
```
Comment (PavelSukhinin) [inline server/src/data/models/abstract/abstract.model.js:1]: how about `knexSnakeCaseMappers` https://vincit.github.io/objection.js/api/objection/#snakecasemappers ?

File: server/src/data/migrations/20211211160459_fill_db.js [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+const getRandomIndex = length => Math.floor(Math.random() * length);
+const mapLinks = images => images.map(image => image.link);
+
+export function up(knex) {
+  return knex.transaction(async trx => {
+    await trx(TableName.USERS).del();
+    await trx(TableName.POSTS).del();
+    await trx(TableName.COMMENTS).del();
+    await trx(TableName.POST_REACTIONS).del();
+    await trx(TableName.IMAGES).del();
```
Comment (PavelSukhinin) [inline server/src/data/migrations/20211211160459_fill_db.js:29]: maybe we shouldn't do it with migration api, but with seed api?

**Review by ghost — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**